### PR TITLE
[Observability] add .slo-observability.* index privileges to built in editor and viewer roles

### PR DESCRIPTION
**Pull request by dominiqueclarke**
Today, the editor and viewer roles do not contain the appropriate index privileges for SLO users.

This PR updates the index privileges to include the `.slo-observability.*` indices.

## Note

I'm not an Elasticsearch dev. I have not ran or tested this or know much about how to operate in Java project. I'd welcome feedback on how to test and collaboration with an Elasticsearch dev to get this over the finish line.

### CODE REVIEW

**Comment by slobodanadamovic** [comment]
@dominiqueclarke 👋 I can help with this PR.

But firstly, I have couple of clarifying questions: 

- I'm wondering if granted index pattern should be `.slo-*` instead of `.slo*`? Not that it matters much (`.slo` is more generic and this is fine), but I just wanted to double check in terms of consistency with [our SLO docs](https://www.elastic.co/guide/en/observability/current/slo-privileges.html).
- Is granting `all` privilege to `editor` role intentional? Would an end-user need to be able to do more than reading and writing to `.slo*` indices? ([full list of all index privileges for reference](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-privileges.html#privileges-list-indices))

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 1]
```diff
@@ -868,6 +868,11 @@ private static RoleDescriptor buildViewerRoleDescriptor() {
                     .indices("/~(([.]|ilm-history-).*)/")
                     .privileges("read", "view_index_metadata")
                     .build(),
+                // Observability
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices(".slo*")
```
Comment (dominiqueclarke) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:873]:
```suggestion
                    .indices(".slo-*")
```

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 2]
```diff
@@ -915,6 +920,10 @@ private static RoleDescriptor buildEditorRoleDescriptor() {
                     .indices("observability-annotations")
                     .privileges("read", "view_index_metadata", "write")
                     .build(),
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices(".slo*")
```
Comment (dominiqueclarke) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:924]:
```suggestion
                    .indices(".slo-*")
```

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 3]
```diff
@@ -868,6 +868,11 @@ private static RoleDescriptor buildViewerRoleDescriptor() {
                     .indices("/~(([.]|ilm-history-).*)/")
                     .privileges("read", "view_index_metadata")
                     .build(),
+                // Observability
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices(".slo-*")
```
Comment (dominiqueclarke) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:873]:
```suggestion
                    .indices(".slo-observability.*")
```

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 4]
```diff
@@ -915,6 +920,10 @@ private static RoleDescriptor buildEditorRoleDescriptor() {
                     .indices("observability-annotations")
                     .privileges("read", "view_index_metadata", "write")
                     .build(),
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices(".slo-*")
```
Comment (dominiqueclarke) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:924]:
```suggestion
                    .indices(".slo-observability.*")
```

**Review by jakelandis — APPROVED** [review]
LGTM

**Pull request merged**